Content.Server/StationEvents/Events/BreakerFlip.cs

```diff
@@ -25,7 +26,16 @@ public override void Started()
     {
         base.Started();
 
-        var allApcs = EntityQuery<ApcComponent>().ToList();
+        if (StationSystem.Stations.Count == 0)
+            return;
+        var chosenStation = RobustRandom.Pick(StationSystem.Stations.ToList());
+
+        var allApcs = EntityQuery<ApcComponent, TransformComponent>().ToList();
+        allApcs = allApcs.FindAll((ent) => 
+        {
+            var (apc, transform) = ent;
+            return apc.MainBreakerEnabled && CompOrNull<StationMemberComponent>(transform.GridUid)?.Station == chosenStation;
+        });
```

Comment:
LINQ is very slow, you're better off iterating it directly yourself.

Reply:
> LINQ is very slow, you're better off iterating it directly yourself.

using foreach?